src/Materials/Textures/mirrorTexture.ts

```diff
@@ -174,7 +174,8 @@ export class MirrorTexture extends RenderTargetTexture {
 
             this._mirrorMatrix.multiplyToRef(this._savedViewMatrix, this._transformMatrix);
 
-            scene.setTransformMatrix(this._transformMatrix, scene.getProjectionMatrix());
+            // Clone to not mark matrices as updated
+            scene.setTransformMatrix(this._transformMatrix.clone(), scene.getProjectionMatrix().clone());

```

Comment:
I don't understand why we need to clone the matrices before passing them?

Reply:
If we use the scene matrices, there is a cache in `setTransformMatrix` : 
```
        if (this._viewUpdateFlag === viewL.updateFlag && this._projectionUpdateFlag === projectionL.updateFlag) {
            return;
        }

        this._viewUpdateFlag = viewL.updateFlag;
        this._projectionUpdateFlag = projectionL.updateFlag;
```

Then in specific cases, it can mess up the regular update of the transform matrix outside of `MirrorTexture`

Reply:
We can also add a parameter `disableCache` in `setTransformMatrix`

Reply:
I still don't get it... Do you have an example where not cloning would fail?

The `scene.setTransformMatrix(this._savedViewMatrix, scene.getProjectionMatrix());` call in `onAfterRenderObservable` should set the right matrix back when the mirror class has finished working.

This pattern is used in several classes (water material, shadow/cascaded shadow generator, reflection probe) and we didn't have problems with it until now. Is there something in this PR that makes this pattern fail?

Reply:
It is connected to prepass with mirror textures.

The call in `onAfterRenderObservable ` is not the problematic one. The problem is the first call to `setTransformMatrix` updates the `updateFlag` cache system on the projection matrix. Then the `PrePassRenderTarget` calls `setTransformMatrix` itself (in `RenderTargetTexture.render`) , which is ineffective because of cache, and causes the post-process to have the wrong projectionMatrix.

The repro is this playground : 
http://localhost:1338/Playground/index-local.html#PIZ1GK#206

Press '2' to get the SSR effect, and change your browser size to alter the projectionMatrix

Reply:
Thanks for the repro.

The problem comes from the fact that the SSAO postprocess is updating the projection matrix (by calling `camera.getProjectionMatrix(true)`) with values depending on the aspect ratio of the RTT, which may not be the same one than the aspect ratio of the output buffer. In the `onAfterRenderObservable` observer of `MirrorTexture`, when we do `scene.setTransformMatrix(this._savedViewMatrix, scene.getProjectionMatrix())`, the projection matrix we get is the last computed one, so it is the one computed by SSAO, which may not have been computed with the right aspect ratio for the output buffer.

Your workaround (cloning the matrices) works because it passes a new matrix which replaces the camera matrix (`scene._projectionMatrix` is normally the same thing than `camera.getProjectionMatrix()`). So, even if SSAO is (indirectly) modifying the camera projection matrix, `scene.getProjectionMatrix()` will return the initial cloned matrix which did not change and has the right values.

Instead of cloning the matrices, I think the right way to fix the problem is to restore the view/projection matrices of the active camera (it's what shadow/cascaded shadow generator and reflection probe are doing) by replacing:
```javascript
scene.setTransformMatrix(this._savedViewMatrix, scene.getProjectionMatrix());
```
by:
```javascript
scene.updateTransformMatrix(); // restore the view/projection matrices of the active camera
```

Reply:
works, and is way cleaner than cloning, thanks a ton !